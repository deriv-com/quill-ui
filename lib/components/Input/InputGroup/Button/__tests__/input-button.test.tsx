import { render, screen, waitFor } from "@testing-library/react";
import InputGroupButton from "..";
import userEvent from "@testing-library/user-event";
import React from "react";

describe("InputGroupButton Component", () => {
    it("should render a default InputGroupButton", () => {
        const { container } = render(
            <InputGroupButton buttonLabel="Submit" label="input" />,
        );

        expect(screen.getByLabelText("input")).toBeInTheDocument();
        expect(screen.getByText("Submit")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("should handle the hover for outline variant", async () => {
        const onHover = jest.fn();
        render(
            <InputGroupButton
                buttonLabel="Button"
                placeholder="Placeholder"
                onMouseEnter={onHover}
                variant="outline"
            />,
        );
        const input = screen.getByPlaceholderText("Placeholder");
        await userEvent.hover(input);
        expect(onHover).toHaveBeenCalledTimes(1);
        expect(input.parentElement).toMatchSnapshot();
    });

    it("disables button when input is empty", () => {
        const { container } = render(<InputGroupButton buttonLabel="Submit" />);

        const button = screen.getByText("Submit").closest("button");
        expect(button).toBeDisabled();
        expect(container).toMatchSnapshot();
    });

    it("enables button when input is not empty", () => {
        const { container } = render(
            <InputGroupButton buttonLabel="Submit" label="input" />,
        );

        const input = screen.getByLabelText("input");
        const button = screen.getByText("Submit");

        userEvent.type(input, "Test");
        expect(button).toBeEnabled();
        expect(container).toMatchSnapshot();
    });

    it("should call buttonCallback if user clicks on button and buttonCallback was passed", async () => {
        const buttonCallback = jest.fn();
        render(
            <InputGroupButton
                buttonLabel="Submit"
                label="input"
                value="User Input"
                buttonCallback={buttonCallback}
            />,
        );

        const button = screen.getByText("Submit");

        userEvent.click(button);
        await waitFor(() => {
            expect(buttonCallback).toHaveBeenCalledTimes(1);
        });
    });
});
