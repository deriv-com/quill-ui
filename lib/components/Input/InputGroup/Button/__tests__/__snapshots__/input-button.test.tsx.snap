// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroupButton Component disables button when input is empty 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--md quill-input__wrapper-with_button-right--md quill-input__wrapper-with_button-right--md--no-label"
    >
      <div
        class="quill-input-icons__wrapper"
      >
        <div
          class="quill-input-no-label__wrapper"
        >
          <input
            button_position="right"
            class="input peer input__align--left input__size--md"
            placeholder=""
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="quill-button quill-button__size--md quill__color--primary-black"
        data-state=""
        disabled=""
      >
        <span
          class="quill-button-label"
        >
          <p
            class="quill-typography quill-typography__body-text__size--sm__weight--bold__decoration--default black"
          >
            Submit
          </p>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`InputGroupButton Component enables button when input is not empty 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--md quill-input__wrapper-with_button-right--md quill-input__wrapper-with_button-right--md--has-label"
    >
      <div
        class="quill-input-icons__wrapper"
      >
        <div
          class="quill-input-label__wrapper"
        >
          <input
            button_position="right"
            class="input peer input__align--left input__size--md"
            id="input"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="label label__status--neutral"
            for="input"
          >
            input
          </label>
        </div>
      </div>
      <button
        class="quill-button quill-button__size--md quill__color--primary-black"
        data-state=""
        disabled=""
      >
        <span
          class="quill-button-label"
        >
          <p
            class="quill-typography quill-typography__body-text__size--sm__weight--bold__decoration--default black"
          >
            Submit
          </p>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`InputGroupButton Component should handle the hover for outline variant 1`] = `
<div
  class="quill-input-no-label__wrapper"
>
  <input
    button_position="right"
    class="input peer input__align--left input__size--md"
    placeholder="Placeholder"
    type="text"
    value=""
  />
</div>
`;

exports[`InputGroupButton Component should render a default InputGroupButton 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--md quill-input__wrapper-with_button-right--md quill-input__wrapper-with_button-right--md--has-label"
    >
      <div
        class="quill-input-icons__wrapper"
      >
        <div
          class="quill-input-label__wrapper"
        >
          <input
            button_position="right"
            class="input peer input__align--left input__size--md"
            id="input"
            placeholder=""
            type="text"
            value=""
          />
          <label
            class="label label__status--neutral"
            for="input"
          >
            input
          </label>
        </div>
      </div>
      <button
        class="quill-button quill-button__size--md quill__color--primary-black"
        data-state=""
        disabled=""
      >
        <span
          class="quill-button-label"
        >
          <p
            class="quill-typography quill-typography__body-text__size--sm__weight--bold__decoration--default black"
          >
            Submit
          </p>
        </span>
      </button>
    </div>
  </div>
</div>
`;
